### PR TITLE
Partitioner: unmount devices recursively before removing

### PR DESCRIPTION
**Pull request by joseivanlopez**
## Problem

In the Expert Partitioner, when a device is deleted, the user is asked right away to unmount the device before deleting it. This is done to avoid possible issues during the commit phase regarding to unmounting devices. Anyway, the user can continue without unmounting. If so, *libstorage-ng* will try to unmount the device during the commit. 

But there are some situations in the Partitioner in which a mounted device is deleted but the user is not asked to unmount it. In general, the unmount question is missing when a device is deleted as side effect of another action. For example:

* When deleting a LVM Logical Volume as side effect of deleting its Volume Group.
* When deleting a partition as side effect of creating a new partition table over its parent device.
* When deleting a device (e.g. MD RAID, Bcache, VG) as side effect of deleting one of its used devices. 
* When deleting a Btrfs subvolume as side effect of deleting its parent file system.
* When deleting a file system as side effect of creating a partition over a directly formatted device.

Related links:

* https://bugzilla.suse.com/show_bug.cgi?id=1171310
* https://trello.com/c/IsCij1Pd/2220-3-ostumbleweed-p5-1171310-partitioner-does-not-ask-to-unmount-btrfs-subvolume

## Solution

Now on, when a device is deleted, all its mounted descendants are asked to be unmounted.

Some code details:

* A new dialog `Y2Partitioner::Dialogs::Unmount` is added (used by the delete actions and also when resizing a file system).
* A new mixing `Y2Partitioner::RecursiveUnmount` is added in order to propose to unmount all mounted devices from a specific device.

## Testing

* Added new unit tests
* Tested manually

## Screenshots

<details>
<summary>Show/Hide</summary>

Screenshot from 2021-02-19 13-28-54

Screenshot from 2021-02-19 14-01-28

</details>

### CODE REVIEW

File: package/yast2-storage-ng.changes [hunk 1]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Fri Feb 19 13:21:35 UTC 2021 - José Iván López González <jlopez@suse.com>
+
+- Partitioner: recursively unmount affected devices when deleting a
```
Comment (dgdavid) [inline package/yast2-storage-ng.changes:4]: NP: I would say it _asks for unmounting_

**Review by dgdavid — APPROVED** [review]
LGTM

File: src/lib/y2partitioner/dialogs/unmount.rb [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      #
+      # @return [String]
+      def content
+        # TRANSLATORS: %{mount_sentences} is replaced by a text describing which devices are mounted.
+        #   Try to keep line breaks.
+        text = format(_("The following devices are currently mounted:\n\n%{mount_sentences}\n\n"),
+          mount_sentences: mount_sentences)
+
+        # TRANSLATORS: %{note} is replaced by a text. Try to keep line breaks.
+        text << format(_("%{note}\n\n"), note: note) if note
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply src/lib/y2partitioner/dialogs/unmount.rb:161]: Yes, translation does not make sense, I would say.

File: src/lib/y2partitioner/dialogs/unmount.rb [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      #
+      # The user can select to unmount, to continue without unmounting (see {#allow_continue?}), or to
+      # cancel.
+      #
+      # The devices are unmounted when the user selects to unmount (see {#unmount_devices}). In case of
+      # unmounting errors, the user is informed by means of another popup.
+      #
+      # @return [:finish, :cancel]
+      def run
+        # FIXME: This check is here to avoid changes in installer behaviour for GA. The dialog for
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply src/lib/y2partitioner/dialogs/unmount.rb:57]: Yes, see https://trello.com/c/WIleYdgI/561-partitioner-improve-popups-when-deleting-things.

**Comment by ancorgs** [comment]
Looking at the screenshots, I'm a little bit worried about the length of the pop-up in ncurses 24x80. For example, in the situation of the second screenshot (12 subvolumes). On one hand, it's true that 12 mount-points is already a extreme case. On the other hand, it would be quite bad if the pop-up gets truncated or something.

**Comment by joseivanlopez** [comment]
> Looking at the screenshots, I'm a little bit worried about the length of the pop-up in ncurses 24x80. For example, in the situation of the second screenshot (12 subvolumes). On one hand, it's true that 12 mount-points is already a extreme case. On the other hand, it would be quite bad if the pop-up gets truncated or something.

I think that `Yast2::Popup` does it work here with auto-wrapping. See:

Screenshot from 2021-02-22 09-21-47

**Pull request merged**